diff -Naur libvorbis-1.2.3/config.sub libvorbis-1.2.3.nacl/config.sub
--- libvorbis-1.2.3/config.sub	2006-08-10 08:17:12.000000000 -0700
+++ libvorbis-1.2.3.nacl/config.sub	2009-10-20 15:01:07.537142000 -0700
@@ -714,6 +714,10 @@
 		basic_machine=i370-ibm
 		os=-mvs
 		;;
+	nacl)
+		basic_machine=i686-pc
+		os=-nacl
+		;;
 	ncr3000)
 		basic_machine=i486-ncr
 		os=-sysv4
@@ -1214,7 +1218,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -nacl*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
