diff -Naur flac-1.2.1/config.sub flac-1.2.1.nacl/config.sub
--- flac-1.2.1/config.sub	2005-04-26 11:23:39.000000000 -0700
+++ flac-1.2.1.nacl/config.sub	2009-10-23 11:03:21.553286000 -0700
@@ -663,6 +663,10 @@
 		basic_machine=i370-ibm
 		os=-mvs
 		;;
+	nacl)
+		basic_machine=i686-pc
+		os=-nacl
+		;;
 	ncr3000)
 		basic_machine=i486-ncr
 		os=-sysv4
@@ -1136,7 +1140,7 @@
 	      | -storm-chaos* | -tops10* | -tenex* | -tops20* | -its* \
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
-	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei*)
+	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -nacl*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -Naur flac-1.2.1/examples/cpp/encode/file/main.cpp flac-1.2.1.nacl/examples/cpp/encode/file/main.cpp
--- flac-1.2.1/examples/cpp/encode/file/main.cpp	2007-09-13 19:58:03.000000000 -0700
+++ flac-1.2.1.nacl/examples/cpp/encode/file/main.cpp	2009-10-23 14:43:06.012399000 -0700
@@ -29,6 +29,7 @@
 #endif
 
 #include <stdio.h>
+#include <string.h>
 #include <stdlib.h>
 #include "FLAC++/metadata.h"
 #include "FLAC++/encoder.h"
diff -Naur flac-1.2.1/src/test_libFLAC++/metadata_manip.cpp flac-1.2.1.nacl/src/test_libFLAC++/metadata_manip.cpp
--- flac-1.2.1/src/test_libFLAC++/metadata_manip.cpp	2007-07-31 15:41:53.000000000 -0700
+++ flac-1.2.1.nacl/src/test_libFLAC++/metadata_manip.cpp	2009-10-23 14:43:06.012399000 -0700
@@ -44,6 +44,9 @@
 #include "test_libs_common/file_utils_flac.h"
 }
 
+/* utime is unimplemented */
+#define utime(path, times) (0)
+
 /******************************************************************************
 	The general strategy of these tests (for interface levels 1 and 2) is
 	to create a dummy FLAC file with a known set of initial metadata
