diff -Naur libpng-1.2.40/config.sub libpng-1.2.40.nacl/config.sub
--- libpng-1.2.40/config.sub	2009-07-11 06:39:02.000000000 -0700
+++ libpng-1.2.40.nacl/config.sub	2009-10-13 12:02:29.808154000 -0700
@@ -759,6 +759,10 @@
 		basic_machine=i370-ibm
 		os=-mvs
 		;;
+	nacl*)
+		basic_machine=i686-pc
+		os=-nacl
+		;;
 	ncr3000)
 		basic_machine=i486-ncr
 		os=-sysv4
@@ -1283,7 +1287,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -nacl*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
